### PR TITLE
[#1407] DG: add Gradle instructions to workflow

### LINKED ISSUE #1407: Add gradle instructions in the workflow section of Developer Guide
**What problem(s) did you run into that caused you to request additional documentation? What, if any, existing documentation relates to this proposal?**

The workflow section in the Developer Guide can contain more specific instructions on back-end testing and check style.

**What would you like added or modified in the documentation?**

I would like to propose to add the relevant gradle instructions to the Developer Guide so that new comers can get started more quickly.

### PR DESCRIPTION
**Pull request by HCY123902**
Resolves #1407 

Commit message:
```
Add gradle instructions for check style and back end testing 
in workflow

The original workflow does not specifically mention the
gradle commands to be executed for style checking and back
end testing.

Adding this information can help new developers to get started
more quickly.

Let's add the relevant gradle instructions to the Developer 
Guide.
```

### CODE REVIEW

File: docs/dg/workflow.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ Our workflow is mostly based on the guidelines given at se-education.org/guides.
 * Make sure you know our coding standards.
   {{ embed('Appendix: Coding Standards', 'styleGuides.md', level=2) }}
 * **Follow [this](https://se-education.org/guides/tutorials/intellijCodeStyle.html) to configure Intellij to follow our coding style**.
-* **This project uses Checkstyle** to check the compliance of Java code. You can use [this document](https://se-education.org/guides/tutorials/checkstyle.html) to find how to use it.
+* **This project uses Checkstyle** to check the compliance of Java code. You can use [this document](https://se-education.org/guides/tutorials/checkstyle.html) to find how to use it. In Particular, run `gradlew checkstyleMain checkstyleTest checkstyleSystemtest` to check the style of all the relevant Java code.
```
Comment (Tejas2805) [inline docs/dg/workflow.md:31]:
```suggestion
* **This project uses Checkstyle** to check the compliance of Java code. You can use [this document](https://se-education.org/guides/tutorials/checkstyle.html) to find how to use it. In particular, run `gradlew checkstyleMain checkstyleTest checkstyleSystemtest` to check the style of all the relevant Java code.
```

**Review by Tejas2805 — APPROVED** [review]
LGTM! One nit.

File: docs/dg/workflow.md (hunk 1 above)
Comment (HCY123902) [reply docs/dg/workflow.md]: Oh thanks for pointing it out. I have edited it.

**Comment by fzdy1914** [comment]
@dcshzj Any idea why the deployment is failing here?

**Comment by dcshzj** [comment]
> @dcshzj Any idea why the deployment is failing here?

The deployment code itself works, but I think the issue was that there were multiple instances of it running. The surge.sh deployment cannot perfectly pinpoint the exact workflow run to download the artifacts for.

One way is to not trigger too many CI runs within a short time frame.

**Comment by fzdy1914** [comment]
> > @dcshzj Any idea why the deployment is failing here?
> 
> The deployment code itself works, but I think the issue was that there were multiple instances of it running. The surge.sh deployment cannot perfectly pinpoint the exact workflow run to download the artifacts for.
> 
> One way is to not trigger too many CI runs within a short time frame. If deployment fails again, we can just trigger it again by closing and reopening the PR.

Can you try to look into a fix for it? As it is affecting our normal usage.

Instead of uploading artifacts into one position. Is modifying the path or name helping, different repo have a different paths?
image

Just look into it first.

**Comment by fzdy1914** [comment]
> > > @dcshzj Any idea why the deployment is failing here?
> > 
> > 
> > The deployment code itself works, but I think the issue was that there were multiple instances of it running. The surge.sh deployment cannot perfectly pinpoint the exact workflow run to download the artifacts for.
> > One way is to not trigger too many CI runs within a short time frame. If deployment fails again, we can just trigger it again by closing and reopening the PR.
> 
> Can you try to look into a fix for it? As it is affecting our normal usage.
> 
> Instead of uploading artifacts into one position. Is modifying the path or name helping, different repo have a different paths?
> image
> 
> Just look into it first.

@dcshzj  And since I will be merging this PR, you may open an issue to keep the discussion.

**Review by fzdy1914 — APPROVED** [review]

**Pull request merged**